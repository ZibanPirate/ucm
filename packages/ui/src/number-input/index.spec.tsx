import { fireEvent, render, screen } from "@testing-library/react";
import { FC, useState } from "react";

import { NumberInput } from ".";

describe(`Testing component '${NumberInput.name}' :`, () => {
  it(`should render with initial value then change it on input change`, () => {
    const FormComponent: FC<{ initialValue: number }> = ({ initialValue }) => {
      const [value, setValue] = useState(initialValue);
      return (
        <NumberInput
          data-testid="number-input"
          label="Mileage"
          value={value}
          onChange={(newValue) => setValue(newValue)}
        />
      );
    };

    const { container } = render(<FormComponent initialValue={20000} />);
    expect(container).toMatchSnapshot("20000");

    const numberInput = screen.getByTestId("number-input");
    fireEvent.change(numberInput, { target: { value: 80000 } });
    expect(container).toMatchSnapshot("80000");
  });

  it(`should render with initial value and no label and no onChange`, () => {
    const { container } = render(<NumberInput data-testid="number-input" value={100} />);
    const numberInput = screen.getByTestId("number-input");
    fireEvent.change(numberInput, { target: { value: 80000 } });
    expect(container).toMatchSnapshot();
  });
});
