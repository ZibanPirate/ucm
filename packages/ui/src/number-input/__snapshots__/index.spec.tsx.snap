// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component 'NumberInput' : should render with initial value and no label and no onChange 1`] = `
<div>
  <div>
    <input
      data-testid="number-input"
      id=""
      name=""
      style="margin: initial;"
      type="number"
      value="80000"
    />
    <label
      for=""
      style="margin: 0.3rem;"
    />
  </div>
</div>
`;

exports[`Testing component 'NumberInput' : should render with initial value then change it on input change: 20000 1`] = `
<div>
  <div>
    <input
      data-testid="number-input"
      id="Mileage"
      name="Mileage"
      style="margin: initial;"
      type="number"
      value="20000"
    />
    <label
      for="Mileage"
      style="margin: 0.3rem;"
    >
      Mileage
    </label>
  </div>
</div>
`;

exports[`Testing component 'NumberInput' : should render with initial value then change it on input change: 80000 1`] = `
<div>
  <div>
    <input
      data-testid="number-input"
      id="Mileage"
      name="Mileage"
      style="margin: initial;"
      type="number"
      value="80000"
    />
    <label
      for="Mileage"
      style="margin: 0.3rem;"
    >
      Mileage
    </label>
  </div>
</div>
`;
